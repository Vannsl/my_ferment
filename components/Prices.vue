<template>
  <section class="bg-gray-100 py-8">
    <div class="container mx-auto px-2 pt-4 pb-12 text-gray-800">
      <h1 class="w-full my-2 text-5xl font-bold leading-tight text-center text-gray-800">Pricing</h1>
      <div class="w-full mb-4">
        <div class="h-1 mx-auto gradient w-64 opacity-25 my-0 py-0 rounded-t"></div>
      </div>
      <div class="flex flex-col sm:flex-row justify-center pt-12 my-12 sm:my-4">
        <price name="Free" price="£0" :limited="true" :list="listFree" />
        <price-highlighted name="Basic" price="x.99" :limited="false" :list="listBasic" />
        <price name="Pro" price="£xx.99" :limited="false" :list="listPro" />
      </div>
    </div>
  </section>
</template>

<script>
import Price from '@/components/Price'
import PriceHighlighted from '@/components/PriceHighlighted'

export default {
  name: 'Prices',
  data() {
    return {
      listFree: ['Thing', 'Thing', 'Thing'],
      listBasic: ['Thing', 'Thing', 'Thing', 'Thing'],
      listPro: ['Thing', 'Thing', 'Thing', 'Thing', 'Thing']
    }
  },
  components: {
    price: Price,
    'price-highlighted': PriceHighlighted
  }
}
</script>
