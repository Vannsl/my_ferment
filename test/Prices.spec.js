import { shallowMount } from '@vue/test-utils'
import Prices from '@/components/Prices'

const factory = () => {
  return shallowMount(Prices)
}

describe('Prices', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
