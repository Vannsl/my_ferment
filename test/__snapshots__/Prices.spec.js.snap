// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prices renders properly 1`] = `"<section class=\\"bg-gray-100 py-8\\"><div class=\\"container mx-auto px-2 pt-4 pb-12 text-gray-800\\"><h1 class=\\"w-full my-2 text-5xl font-bold leading-tight text-center text-gray-800\\">Pricing</h1> <div class=\\"w-full mb-4\\"><div class=\\"h-1 mx-auto gradient w-64 opacity-25 my-0 py-0 rounded-t\\"></div></div> <div class=\\"flex flex-col sm:flex-row justify-center pt-12 my-12 sm:my-4\\"><price-stub name=\\"Free\\" price=\\"£0\\" limited=\\"true\\" list=\\"Thing,Thing,Thing\\"></price-stub> <price-highlighted-stub name=\\"Basic\\" price=\\"x.99\\" list=\\"Thing,Thing,Thing,Thing\\"></price-highlighted-stub> <price-stub name=\\"Pro\\" price=\\"£xx.99\\" list=\\"Thing,Thing,Thing,Thing,Thing\\"></price-stub></div></div></section>"`;
